src/UglyToad.PdfPig/Geometry/PdfPoint.cs

```diff
@@ -82,6 +82,39 @@ internal PdfVector ToVector()
             return new PdfVector(X, Y);
         }
 
+        /// <summary>
+        /// Converts this <see cref="PdfPoint"/> into an array.
+        /// </summary>
+        /// <returns></returns>
+        public double[] ToDouble()
```

Comment:
I wonder if this would be better as a `ValueTuple` rather than an array. Since the array will always be known to have 2 members a tuple has a couple of benefits:
+ Prevents incorrect indexing by consumers
+ Avoids an allocation on the heap which needs garbage collection

This would look something like:
```suggestion
        /// <summary>Converts this <see cref="PdfPoint" to double values.<.summary>
        public (double x, double y) ToDouble()
```

I think PdfPig already includes ValueTuple as a dependency (it's a nuget package for earlier versions of the .NET Framework) so all consumers will be able to use it.

(secondary nitpick on the empty `<returns>` tags).

Reply:
On this, I would really need a double array for computation purpose...

Reply:
Hmm, I'd be tempted to leave this off the public API for now. My gut instinct is that it'll confuse consumers of the library over the alternative of having a tuple. You could add:

````
public static class PdfPointExtensions
{
    public static double[] AsDoubleArray(this PdfPoint point) => new [] { (double)point.X, (double)point.Y };
}
```

Either to your consumer code or to PdfPig (or internally to PdfPig if it's for layout analysis code internally) but I'd avoid having it as a method directly on the public API of `PdfPoint` for now to reduce the amount of autocomplete options.

Reply:
okay, you are right. Let's keep it simple and leave it put for the moment